llvmlite/binding/value.py

```diff
@@ -140,6 +186,7 @@ def add_function_attribute(self, attr):
         attr : str
             attribute name
         """
+        assert self.is_function
```

Comment:
I think it's better to explicitly raise an exception instead of using `assert` in case someone runs with `python -O` (assertions optimized out).

Reply:
Agreed and fixed.